### DISCUSSION TITLE
Execute Vehicle Action - Summary

### DISCUSSION THREAD

**Discussion by Kolbi** [audiconnect/audi_connect_ha#282] [General]
Working services:

Not working services:
            - start_preheater https://github.com/audiconnect/audi_connect_ha/issues/262
            - stop_preheater https://github.com/audiconnect/audi_connect_ha/issues/262
            - lock https://github.com/audiconnect/audi_connect_ha/issues/262
            - unlock https://github.com/audiconnect/audi_connect_ha/issues/262
            - start_climatisation https://github.com/audiconnect/audi_connect_ha/issues/267 https://github.com/audiconnect/audi_connect_ha/issues/264
            - start_charger https://github.com/audiconnect/audi_connect_ha/issues/227
            - stop_charger https://github.com/audiconnect/audi_connect_ha/issues/227

Not tested yet:
            - stop_climatisation
            - start_timed_charger
            - start_preheater
            - stop_preheater
            - start_window_heating
            - stop_window_heating

Reminder: Maybe vehicle depended APIs? See https://github.com/cyr-ius/audiconnectpy

**Comment by t0bias-r** [comment]
Endpoint for wakeup:
https://emea.bff.cariad.digital/vehicle/v1/vehicles/WAUZZZGE3RB987654/vehiclewakeup

Response:
`{
    "data": {
        "requestID": "9757aad4-d40e-4a63-aede-6fa878a1e6d9"
    }
}`


Then polling the pending requests.
The poll interval of the app seems to be 5 seconds
URL: https://emea.bff.cariad.digital/vehicle/v1/vehicles/WAUZZZGE3RB987654/pendingrequests

Response (in progress):
`{
    "data": [
        {
            "id": "9757aad4-d40e-4a63-aede-6fa878a1e6d9",
            "userId": "935c28cb-d14a-416f-8a85-27fbd11b0688",
            "vin": "WAUZZZGE3RB987654",
            "httpMethod": "POST",
            "path": "/vehicles/WAUZZZGE3RB987654/vehiclewakeup",
            "capability": "vehicleWakeUp",
            "operation": "wakeup",
            "operationGroup": "vehicleWakeUp",
            "body": "{\"callback\":\"https://emea.bff.cariad.digital/vehicle/v1/vehicles/callbacks/WAUZZZGE3RB987654/vehicleWakeUp/wakeup/POST\"}",
            "vcfRequestId": "1937887907",
            "status": "in_progress",
            "queuedOperation": null,
            "timestamp": "2024-03-05T10:44:39.756179479Z"
        }
    ]
}`
Response (successful - just status has changed):
`{
    "data": [
        {
            "id": "9757aad4-d40e-4a63-aede-6fa878a1e6d9",
            "userId": "935c28cb-d14a-416f-8a85-27fbd11b0688",
            "vin": "WAUZZZGE3RB987654",
            "httpMethod": "POST",
            "path": "/vehicles/WAUZZZGE3RB987654/vehiclewakeup",
            "capability": "vehicleWakeUp",
            "operation": "wakeup",
            "operationGroup": "vehicleWakeUp",
            "body": "{\"callback\":\"https://emea.bff.cariad.digital/vehicle/v1/vehicles/callbacks/WAUZZZGE3RB987654/vehicleWakeUp/wakeup/POST\"}",
            "vcfRequestId": "1937887907",
            "status": "successful",
            "queuedOperation": null,
            "timestamp": "2024-03-05T10:44:39.756179479Z"
        }
    ]
}`

**Comment by t0bias-r** [comment]
Endpoint for climater (seems to be unchanged):
https://mal-3a.prd.eu.dp.vwg-connect.com/api/bs/climatisation/v1/vehicles/WAUZZZGE3RB987654/climater/actions
POST for Stop:
`{
    "action": {
        "type": "stopClimatisation"
    }
}`

POST for Start:
`{"action":{"type":"startClimatisation","settings":{"targetTemperature":2950,"heaterSource":"electric","climaterElementSettings":{"isMirrorHeatingEnabled":false,"zoneSettings":{"zoneSetting":[{"value":{"position":"frontLeft","isEnabled":true}},{"value":{"position":"frontRight","isEnabled":false}}]}}}}}`

Note: Temperature is in 0.1 Kelvin and zoneSettings are not available for every car.

Response Stop:
`{
    "action": {
        "type": "stopClimatisation",
        "actionId": 627905555,
        "actionState": "queued"
    }
}`

Response Start:
`{
    "action": {
        "type": "startClimatisation",
        "actionId": 627894444,
        "actionState": "queued",
        "settings": {
            "targetTemperature": 2950,
            "climatisationWithoutHVpower": true,
            "heaterSource": "electric",
            "climaterElementSettings": {
                "isMirrorHeatingEnabled": false,
                "zoneSettings": {
                    "zoneSetting": [
                        {
                            "value": {
                                "isEnabled": true,
                                "position": "frontLeft"
                            }
                        },
                        {
                            "value": {
                                "isEnabled": false,
                                "position": "frontRight"
                            }
                        }
                    ]
                }
            }
        }
    }
}`


Polling result:
https://mal-3a.prd.eu.dp.vwg-connect.com/api/bs/climatisation/v1/vehicles/WAUZZZGE3RB987654/climater/actions/627894444
`{
    "action": {
        "type": "startClimatisation",
        "actionId": 627894444,
        "actionState": "succeeded"
    }
}`

**Comment by t0bias-r** [comment]
Endpoint for capabilities:
https://emea.bff.cariad.digital/vehicle/v1/vehicles/WAUZZZGE3RB987654/capabilities

`{
    "vin": "WAUZZZGE3RB987654",
    "capabilities": {
        "fuelStatus": {
            "id": "fuelStatus",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {
                "getFuelStatus": {
                    "id": "getFuelStatus",
                    "scopes": [
                        "fuelLevels"
                    ]
                }
            },
            "parameters": []
        },
        "predictiveRouting": {
            "id": "predictiveRouting",
            "expirationDate": "2026-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "tripStatistics": {
            "id": "tripStatistics",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {
                "deleteTripdataShortterm": {
                    "id": "deleteTripdataShortterm",
                    "scopes": [
                        "tripStatistics"
                    ]
                },
                "getTripdataShortterm": {
                    "id": "getTripdataShortterm",
                    "scopes": [
                        "tripStatistics"
                    ]
                },
                "deleteTripdataCyclic": {
                    "id": "deleteTripdataCyclic",
                    "scopes": [
                        "tripStatistics"
                    ]
                },
                "getTripdataCyclic": {
                    "id": "getTripdataCyclic",
                    "scopes": [
                        "tripStatistics"
                    ]
                },
                "deleteTripdataLongterm": {
                    "id": "deleteTripdataLongterm",
                    "scopes": [
                        "tripStatistics"
                    ]
                },
                "getTripdataLongterm": {
                    "id": "getTripdataLongterm",
                    "scopes": [
                        "tripStatistics"
                    ]
                },
                "getTripdataLongtermLast": {
                    "id": "getTripdataLongtermLast",
                    "scopes": [
                        "tripStatistics"
                    ]
                },
                "deleteTripdataLongtermByID": {
                    "id": "deleteTripdataLongtermByID",
                    "scopes": [
                        "tripStatistics"
                    ]
                },
                "getTripdataCyclicLast": {
                    "id": "getTripdataCyclicLast",
                    "scopes": [
                        "tripStatistics"
                    ]
                },
                "deleteTripdataCyclicByID": {
                    "id": "deleteTripdataCyclicByID",
                    "scopes": [
                        "tripStatistics"
                    ]
                },
                "getTripdataShorttermLast": {
                    "id": "getTripdataShorttermLast",
                    "scopes": [
                        "tripStatistics"
                    ]
                },
                "deleteTripdataShorttermByID": {
                    "id": "deleteTripdataShorttermByID",
                    "scopes": [
                        "tripStatistics"
                    ]
                }
            },
            "parameters": []
        },
        "vehicleWakeUpTrigger": {
            "id": "vehicleWakeUpTrigger",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {
                "postVehiclewakeupUpdate": {
                    "id": "postVehiclewakeupUpdate",
                    "scopes": []
                }
            },
            "parameters": []
        },
        "wifiHotspot": {
            "id": "wifiHotspot",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "chargingTimers": {
            "id": "chargingTimers",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {
                "getChargingTimers": {
                    "id": "getChargingTimers",
                    "scopes": [
                        "charging"
                    ]
                },
                "putChargingTimers": {
                    "id": "putChargingTimers",
                    "scopes": [
                        "manageCharging"
                    ]
                }
            },
            "parameters": [
                {
                    "key": "timerStepSizeInMinutes",
                    "value": "5"
                },
                {
                    "key": "supportsSingleTimer",
                    "value": "false"
                },
                {
                    "key": "supportsRepeatingTimer",
                    "value": "true"
                },
                {
                    "key": "allowedSchedulingIntervalInDays",
                    "value": "365"
                },
                {
                    "key": "multipleActiveTimersAllowed",
                    "value": "true"
                },
                {
                    "key": "timeZoneReference",
                    "value": "vehicleLocal"
                }
            ]
        },
        "trafficLights": {
            "id": "trafficLights",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": false,
            "status": [
                "MissingLicense"
            ],
            "operations": {},
            "parameters": []
        },
        "trafficSigns": {
            "id": "trafficSigns",
            "expirationDate": "2026-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "dataPlan": {
            "id": "dataPlan",
            "expirationDate": "2122-03-24T00:00:00Z",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "eRoutePlanner": {
            "id": "eRoutePlanner",
            "expirationDate": "2026-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "functionOnDemand": {
            "id": "functionOnDemand",
            "expirationDate": "2122-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "onlineSpeech": {
            "id": "onlineSpeech",
            "expirationDate": "2026-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "vehicleHealthCampaigns": {
            "id": "vehicleHealthCampaigns",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "charging": {
            "id": "charging",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {
                "getChargingMode": {
                    "id": "getChargingMode",
                    "scopes": [
                        "charging"
                    ]
                },
                "putChargingSettings": {
                    "id": "putChargingSettings",
                    "scopes": [
                        "manageCharging"
                    ]
                },
                "getChargingStatus": {
                    "id": "getChargingStatus",
                    "scopes": [
                        "charging"
                    ]
                },
                "postChargingStart": {
                    "id": "postChargingStart",
                    "scopes": [
                        "manageCharging"
                    ]
                },
                "postChargingStop": {
                    "id": "postChargingStop",
                    "scopes": [
                        "manageCharging"
                    ]
                },
                "getChargingCareStatus": {
                    "id": "getChargingCareStatus",
                    "scopes": [
                        "manageCharging"
                    ]
                },
                "putChargingMode": {
                    "id": "putChargingMode",
                    "scopes": [
                        "manageCharging"
                    ]
                },
                "getChargingRequestsByID": {
                    "id": "getChargingRequestsByID",
                    "scopes": [
                        "manageCharging"
                    ]
                },
                "getChargingSettings": {
                    "id": "getChargingSettings",
                    "scopes": [
                        "charging"
                    ]
                }
            },
            "parameters": [
                {
                    "key": "allowPlugUnlockACPermanent",
                    "value": "false"
                },
                {
                    "key": "allowPlugUnlockACOnce",
                    "value": "false"
                },
                {
                    "key": "allowPlugUnlockDCPermanent",
                    "value": "true"
                },
                {
                    "key": "allowPlugUnlockDCOnce",
                    "value": "false"
                },
                {
                    "key": "supportsTargetStateOfCharge",
                    "value": "true"
                },
                {
                    "key": "chargeCurrentMaximumAllowedValue",
                    "value": ""
                },
                {
                    "key": "targetStateOfChargeMinimumAllowedValue",
                    "value": "50"
                },
                {
                    "key": "targetStateOfChargeStepSize",
                    "value": "10"
                }
            ]
        },
        "plugAndCharge": {
            "id": "plugAndCharge",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {
                "getPlugandchargeStatusContract": {
                    "id": "getPlugandchargeStatusContract",
                    "scopes": [
                        "plugAndCharge"
                    ]
                },
                "postPlugandchargeUninstall": {
                    "id": "postPlugandchargeUninstall",
                    "scopes": [
                        "plugAndCharge"
                    ]
                },
                "postPlugandchargeUpdate": {
                    "id": "postPlugandchargeUpdate",
                    "scopes": [
                        "plugAndCharge"
                    ]
                },
                "postPlugandchargeActivate": {
                    "id": "postPlugandchargeActivate",
                    "scopes": [
                        "plugAndCharge"
                    ]
                },
                "postPlugandchargeDeactivate": {
                    "id": "postPlugandchargeDeactivate",
                    "scopes": [
                        "plugAndCharge"
                    ]
                },
                "postPlugandchargeInstall": {
                    "id": "postPlugandchargeInstall",
                    "scopes": [
                        "plugAndCharge"
                    ]
                },
                "getPlugandchargeRequestsByID": {
                    "id": "getPlugandchargeRequestsByID",
                    "scopes": [
                        "plugAndCharge"
                    ]
                },
                "getPlugandchargeStatus": {
                    "id": "getPlugandchargeStatus",
                    "scopes": [
                        "plugAndCharge"
                    ]
                }
            },
            "parameters": []
        },
        "vehicleHealthWarnings": {
            "id": "vehicleHealthWarnings",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {
                "getWarninglightsLast": {
                    "id": "getWarninglightsLast",
                    "scopes": [
                        "warning_lights"
                    ]
                }
            },
            "parameters": []
        },
        "googleEarth": {
            "id": "googleEarth",
            "expirationDate": "2026-03-24T00:00:00Z",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "webAppWeather": {
            "id": "webAppWeather",
            "expirationDate": "2026-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "destinationSync": {
            "id": "destinationSync",
            "expirationDate": "2026-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "vehicleLights": {
            "id": "vehicleLights",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {
                "getLightsStatus": {
                    "id": "getLightsStatus",
                    "scopes": [
                        "vehicleLights"
                    ]
                }
            },
            "parameters": []
        },
        "onlineTrafficPlus": {
            "id": "onlineTrafficPlus",
            "expirationDate": "2026-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "poiSearch": {
            "id": "poiSearch",
            "expirationDate": "2026-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "amazonMusic": {
            "id": "amazonMusic",
            "expirationDate": "2026-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "chargingProfiles": {
            "id": "chargingProfiles",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {
                "deleteChargingProfilesByID": {
                    "id": "deleteChargingProfilesByID",
                    "scopes": [
                        "charging"
                    ]
                },
                "getChargingProfiles": {
                    "id": "getChargingProfiles",
                    "scopes": [
                        "charging"
                    ]
                },
                "postChargingProfiles": {
                    "id": "postChargingProfiles",
                    "scopes": [
                        "manageCharging"
                    ]
                },
                "putChargingProfiles": {
                    "id": "putChargingProfiles",
                    "scopes": [
                        "manageCharging"
                    ]
                }
            },
            "parameters": [
                {
                    "key": "supportsMaxChargingCurrent",
                    "value": "false"
                },
                {
                    "key": "allowPlugUnlockACPermanent",
                    "value": "false"
                },
                {
                    "key": "allowPlugUnlockACOnce",
                    "value": "false"
                },
                {
                    "key": "allowPlugUnlockDCPermanent",
                    "value": "false"
                },
                {
                    "key": "allowPlugUnlockDCOnce",
                    "value": "false"
                },
                {
                    "key": "minimumStateOfChargeFixedValue",
                    "value": "25"
                },
                {
                    "key": "minimumStateOfChargeMinimumAllowedValue",
                    "value": ""
                },
                {
                    "key": "minimumStateOfChargeMaximumAllowedValue",
                    "value": ""
                },
                {
                    "key": "minimumStateOfChargeStepSize",
                    "value": ""
                },
                {
                    "key": "numberOfPreferredChargingTimes",
                    "value": "1"
                },
                {
                    "key": "numberOfTimers",
                    "value": "0"
                },
                {
                    "key": "supportsRepeatingTimer",
                    "value": "false"
                },
                {
                    "key": "supportsSingleTimer",
                    "value": "false"
                },
                {
                    "key": "supportsUsePrivateCurrent",
                    "value": "false"
                },
                {
                    "key": "minimumStateOfChargeSupportsDisabling",
                    "value": "true"
                },
                {
                    "key": "allowLocationModification",
                    "value": "true"
                },
                {
                    "key": "radiusInMeters",
                    "value": "100"
                },
                {
                    "key": "supportsTimerClimatisation",
                    "value": "false"
                },
                {
                    "key": "supportsVehiclePositionedInProfileID",
                    "value": "true"
                },
                {
                    "key": "targetStateOfChargeStepSize",
                    "value": "10"
                },
                {
                    "key": "numberOfProfiles",
                    "value": "7"
                },
                {
                    "key": "targetStateOfChargeMinimumAllowedValue",
                    "value": "50"
                },
                {
                    "key": "supportsTargetStateOfCharge",
                    "value": "true"
                },
                {
                    "key": "timeZoneReference",
                    "value": "vehicleLocal"
                }
            ]
        },
        "emergencyCalling": {
            "id": "emergencyCalling",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "climatisationTimers": {
            "id": "climatisationTimers",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {
                "getClimatisationTimers": {
                    "id": "getClimatisationTimers",
                    "scopes": [
                        "climatisation"
                    ]
                },
                "putClimatisationTimers": {
                    "id": "putClimatisationTimers",
                    "scopes": [
                        "manageClimatisation"
                    ]
                }
            },
            "parameters": [
                {
                    "key": "timerStepSizeInMinutes",
                    "value": "5"
                },
                {
                    "key": "supportsSingleTimer",
                    "value": "true"
                },
                {
                    "key": "supportsRepeatingTimer",
                    "value": "false"
                },
                {
                    "key": "allowedSchedulingIntervalInDays",
                    "value": "7"
                },
                {
                    "key": "multipleActiveTimersAllowed",
                    "value": "true"
                },
                {
                    "key": "timeZoneReference",
                    "value": "vehicleLocal"
                }
            ]
        },
        "dealerAppointment": {
            "id": "dealerAppointment",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": false,
            "status": [
                "InitiallyDisabled",
                "DisabledByUser"
            ],
            "operations": {
                "getMaintenanceStatus": {
                    "id": "getMaintenanceStatus",
                    "scopes": [
                        "serviceInterval"
                    ]
                },
                "getPredictivemaintenanceDealerappointment": {
                    "id": "getPredictivemaintenanceDealerappointment",
                    "scopes": [
                        "serviceInterval"
                    ]
                },
                "getWarninglightsLast": {
                    "id": "getWarninglightsLast",
                    "scopes": [
                        "warning_lights"
                    ]
                }
            },
            "parameters": []
        },
        "vehicleHealthInspection": {
            "id": "vehicleHealthInspection",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {
                "getMaintenanceStatus": {
                    "id": "getMaintenanceStatus",
                    "scopes": [
                        "serviceInterval"
                    ]
                }
            },
            "parameters": []
        },
        "vehicleHealthWakeUp": {
            "id": "vehicleHealthWakeUp",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {
                "getWarninglightsLast": {
                    "id": "getWarninglightsLast",
                    "scopes": [
                        "warning_lights"
                    ]
                }
            },
            "parameters": []
        },
        "hybridRadio": {
            "id": "hybridRadio",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "localHazards": {
            "id": "localHazards",
            "expirationDate": "2026-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "parkingPosition": {
            "id": "parkingPosition",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {
                "getParkingposition": {
                    "id": "getParkingposition",
                    "scopes": [
                        "parking_position"
                    ]
                }
            },
            "parameters": []
        },
        "onlineLogBook": {
            "id": "onlineLogBook",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": false,
            "status": [
                "MissingLicense"
            ],
            "operations": {},
            "parameters": []
        },
        "transactionHistoryLockUnlock": {
            "id": "transactionHistoryLockUnlock",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {
                "getTransactionhistoryAccess": {
                    "id": "getTransactionhistoryAccess",
                    "scopes": [
                        "messageCenter"
                    ]
                }
            },
            "parameters": []
        },
        "webAppPoiSearch": {
            "id": "webAppPoiSearch",
            "expirationDate": "2026-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "roadsideAssistant": {
            "id": "roadsideAssistant",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "measurements": {
            "id": "measurements",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {
                "getMeasurements": {
                    "id": "getMeasurements",
                    "scopes": [
                        "range",
                        "mileage"
                    ]
                }
            },
            "parameters": []
        },
        "state": {
            "id": "state",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {
                "getAccessStatus": {
                    "id": "getAccessStatus",
                    "scopes": [
                        "doors_windows"
                    ]
                }
            },
            "parameters": []
        },
        "webRadio": {
            "id": "webRadio",
            "expirationDate": "2122-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "theming": {
            "id": "theming",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": [
                {
                    "key": "themeFeatureSet",
                    "value": ""
                }
            ]
        },
        "vehicleWakeUp": {
            "id": "vehicleWakeUp",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {
                "postVehiclewakeup": {
                    "id": "postVehiclewakeup",
                    "scopes": []
                },
                "getVehiclewakeupRequestsByID": {
                    "id": "getVehiclewakeupRequestsByID",
                    "scopes": []
                }
            },
            "parameters": []
        },
        "bluetoothIdent": {
            "id": "bluetoothIdent",
            "expirationDate": "2122-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "onStreetParking": {
            "id": "onStreetParking",
            "expirationDate": "2026-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "onlineTraffic": {
            "id": "onlineTraffic",
            "expirationDate": "2026-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "access": {
            "id": "access",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {
                "getAccessRequestsByID": {
                    "id": "getAccessRequestsByID",
                    "scopes": [
                        "lock_unlock"
                    ]
                },
                "postAccessUnlock": {
                    "id": "postAccessUnlock",
                    "scopes": [
                        "lock_unlock"
                    ]
                },
                "postAccessUnlockwithoutspin": {
                    "id": "postAccessUnlockwithoutspin",
                    "scopes": [
                        "lock_unlock"
                    ]
                },
                "postAccessLock": {
                    "id": "postAccessLock",
                    "scopes": [
                        "lock_unlock"
                    ]
                },
                "postAccessLockwithoutspin": {
                    "id": "postAccessLockwithoutspin",
                    "scopes": [
                        "lock_unlock"
                    ]
                }
            },
            "parameters": []
        },
        "mapUpdateSd": {
            "id": "mapUpdateSd",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "gracenote": {
            "id": "gracenote",
            "expirationDate": "2122-03-24T00:00:00Z",
            "userDisablingAllowed": false,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        },
        "onlineCarCare": {
            "id": "onlineCarCare",
            "expirationDate": "2033-03-24T00:00:00Z",
            "userDisablingAllowed": true,
            "endpoint": "vs",
            "isEnabled": true,
            "status": [],
            "operations": {},
            "parameters": []
        }
    },
    "parameters": {}
}`

**Comment by t0bias-r** [comment]
Immediate charging:
https://emea.bff.cariad.digital/vehicle/v1/vehicles/WAUZZZGE3RB987654/charging/mode

POST:
`{
    "preferredChargeMode": "manual"
}`

Response:
`{
    "data": {
        "requestID": "fcbd501d-38d8-435a-a801-dd456e4b22b9"
    }
}`



Polling status:
https://emea.bff.cariad.digital/vehicle/v1/vehicles/WAUZZZGE3RB987654/pendingrequests
`{
    "data": [
        {
            "id": "fcbd501d-38d8-435a-a801-dd456e4b22b9",
            "userId": "935c28cb-d14a-416f-8a85-27fbd11b0688",
            "vin": "WAUZZZGE3RB987654",
            "httpMethod": "PUT",
            "path": "/vehicles/WAUZZZGE3RB987654/charging/mode",
            "capability": "charging",
            "operation": "mode",
            "operationGroup": "chargingMode",
            "body": "{\"callback\":\"https://emea.bff.cariad.digital/vehicle/v1/vehicles/callbacks/WAUZZZGE3RB987654/charging/mode/PUT\",\"preferredChargeMode\":\"manual\"}",
            "vcfRequestId": "627947777",
            "status": "in_progress",
            "queuedOperation": null,
            "timestamp": "2024-03-12T20:01:35.038461731Z"
        }
    ]
}`

**Comment by DutchKillerbee** [comment]
Hello @Kolbi @t0bias-r 
I would like to help to make the start_charging and stop_charging work again.
What can I do to help?
I have minor (python) programming skills. I have a test system available where I can change the code in the custom_components directory.
Please let me know what I can do.

**Comment by coreywillwhat** [comment]
@t0bias-r How are you getting the Endpoints? I'm in US so if I can compare mine against what they should be, and against yours, I maybe be able to figure out the updates to make.

**Reply by Kolbi** [reply]
I guess sniffing the mobilephone app was his way to retrieve the information about the APIs

**Reply by t0bias-r** [reply]
I use Frida and Fiddler to trace the network traffic of the myAudi app on an Android smartphone.

Frida is needed to bypass the certificate pinning of the myAudi app. Otherwise you're not able to decrypt the traffic.
I patch and repack the Android myAudi apk and sideload it to be able to connect with Frida over USB (https://koz.io/using-frida-on-android-without-root/).

Fiddler is then configured as a MITM proxy for the WiFi connection. It records all network traffic and decrypts it.

**Reply by coreywillwhat** [reply]
Thanks for the info! is root required or does the patch you put in the APK allow you to bypass that?

**Reply by t0bias-r** [reply]
root on Android is not required when APK is patched - see the link above

**Reply by coreywillwhat** [reply]
totally spaced on seeing that. Thank you, I appreciate it.

**Comment by DutchKillerbee** [comment]
Start Stop charger seems to work for me (version 1.6.2). I disabled pin code protection (of my audi account in the car)
Maybe there is no relation but just to let you know

**Comment by coreywillwhat** [comment]
@t0bias-r I've got climate control/lock/unlock updated and working for the new US endpoints. I would venture to guess the endpoints have changed for EU Region as well. 

Are you able to provide the endpoints for EU again, for climater start/stop and lock/unlock, as well as all of the POST request, responses and polling for updates? If there are any other services you could provide as well it would be much appreciated. The more data you can provide the better. Thanks so much!

**Comment by smartathome** [comment]
Hi, would there be a way to remotely enable/disable the warningLights? From the MyAudi app this is currently possible.
